src/gluonts/dataset/tabular.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Iterator, List, Union, Dict
+
+import pandas as pd
+
+from gluonts.dataset.common import Dataset, DataEntry, ProcessDataEntry
+from gluonts.dataset.field_names import FieldName
+
+
+@dataclass
+class DataFramesDataset(Dataset):
```

Comment:
Couldn't we name this `TabularDataset`?

Isn't the dataframe aspect an internal implementation detail?

Reply:
> Isn't the dataframe aspect an internal implementation detail?

The object is constructed by giving it a `DataFrame`, so...

Reply:
I think that's not a very good argument -- we generally don't just call things by what we pass them as arguments.

Why call the module `tabular`, if it exposes this class?

Reply:
I disagree: what is being provided here is a very specific type of dataset, which sources data from a pandas dataframe; if anything, `FileDataset` is a misnomer since it doesn’t support just “any file” but a specific format (JSON lines).

The module could be renamed to something else: `pandas` or `dataframe`?